### DISCUSSION TITLE
RandZoom Transform with different interpolation for mask and image

### DISCUSSION THREAD

**Discussion by OmarAshkar** [Project-MONAI/MONAI#6889] [Q&A] [answered]
Hello, 

I am wondering if I can use the RandZoom with different interpolation, cause for the mask, it should stay as 0 or 1. but for image, the scale is allowed to interpolate. So if the mode changed from one line to the next, the randomness should be fixed if that makes sense.

Thank you,
Omar

**Comment by wyli** [comment]
yes, similar demo cases are described here https://github.com/Project-MONAI/tutorials/blob/main/modules/3d_image_transforms.ipynb 

(the tutorial/modules folder has quite a few examples about the transforms in general https://github.com/Project-MONAI/tutorials/tree/main/modules)

**Reply by OmarAshkar** [reply]
thanks alot @wyli. Just additional enquiry. I need to augment the data, keeping the original and augmented. When I test and apply the transforms I don't get more that the original length

```
train_transforms = Compose(
    [
        LoadImaged(keys=["img", "seg"]),
        EnsureChannelFirstd(keys=["img", "seg"]),
        ScaleIntensityd(keys=["img", "seg"]),
        # RandCropByPosNegLabeld(
        #     keys=["img", "seg"], label_key="seg", spatial_size=[96, 96], pos=1, neg=1, num_samples=4
        # ),
        RandRotate90d(keys=["img", "seg"], prob=0.5, spatial_axes=[0, 1]),
        RandRotate90d(keys=["img", "seg"], prob=0.5, spatial_axes=[0, 1]),
    ]
)
len(train_files) # 3

x  = train_transforms(train_files)
len(x) # 3
```

Also as you see I repeated the RandRotate90d 2 times in sense to generate 2 samples. So the desired end should be 5 here if I understand correctly right?

**Reply by wyli** [reply, marked as answer]
Sure, the code looks fine, the random augmentations are 'in-place' data updates, different calls would output different augmented versions of the original inputs, this is often used with a dataset and dataloader. Please see https://github.com/Project-MONAI/monai-bootcamp/blob/main/MONAICore/Intro%20to%20MONAI.ipynb
